src/client_metrics.rs

```diff
@@ -284,34 +291,39 @@ impl ImpactMetricEnv {
     }
 }
 
-impl Merge for ImpactMetricEnv {
-    fn merge(self, other: ImpactMetricEnv) -> ImpactMetricEnv {
-        let mut result = self;
-        let mut samples_by_labels: HashMap<String, MetricSample> = HashMap::new();
-        let is_counter = result.impact_metric.r#type == MetricType::Counter
-            && other.impact_metric.r#type == MetricType::Counter;
-
-        for sample in &result.impact_metric.samples {
-            let labels_key = sample.labels_to_key();
-            samples_by_labels.insert(labels_key, sample.clone());
-        }
-
-        for sample in other.impact_metric.samples {
-            let labels_key = sample.labels_to_key();
-            if is_counter {
-                if let Some(existing_sample) = samples_by_labels.get_mut(&labels_key) {
-                    existing_sample.value += sample.value;
+impl MergeMut for ImpactMetricEnv {
+    fn merge(&mut self, other: ImpactMetricEnv) {
+        let is_counter = self.impact_metric.r#type == MetricType::Counter;
+
+        self.impact_metric
+            .samples
+            .extend(other.impact_metric.samples);
+        self.impact_metric
```

Comment:
BTreeMap takes away a lot of the heavy lifting here. We don't have to emit String keys, which reduces malloc pressure